src/main/scala/com/example/solution/http/json.scala

```diff
@@ -0,0 +1,67 @@
+package com.example.solution.http
+
+import cats.Applicative
+import com.example.solution.domain.movie.{Movie, MovieId, MovieTitle}
+import com.example.solution.domain.room.{Room, RoomId}
+import com.example.solution.domain.screening.{Screening, ScreeningId}
+import com.example.solution.dto.reservation.{MadeReservation, PickedSeat, ReservationData, ScreeningInterval}
+import com.example.solution.dto.screening.{AvailableSeat, RoomData, ScreeningData}
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+import org.http4s.EntityEncoder
+import org.http4s.circe.jsonEncoderOf
+import cats.syntax.all._
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+object json extends JsonCodecs {
+  implicit def deriveEntityEncoder[F[_] : Applicative, A: Encoder]: EntityEncoder[F, A] = jsonEncoderOf[F, A]
+}
+
+
+private[http] trait JsonCodecs {
+
+  implicit val MovieIdDecoder: Decoder[MovieId] = deriveDecoder[MovieId]
```

Comment:
derive codecs are convenient. Do they have any downsides?

Reply:
I know automatic derivation might be problematic, I don't know what would be a downside of using semi-automatic derivation. Sometimes there are many ways of encoding/decoding a data type, so we might have to write it ourselves(I wrote one for LocalDateTime)